src/cascadia/TerminalConnection/CTerminalHandoff.cpp

```diff
@@ -0,0 +1,86 @@
+#include "pch.h"
+
+#include "CTerminalHandoff.h"
+
+// The callback function when a connection is received
+static NewHandoff _pfnHandoff = nullptr;
+// The registration ID of the class object for clean up later
+static DWORD g_cTerminalHandoffRegistration = 0;
+
+// Routine Description:
+// - Starts listening for TerminalHandoff requests by registering
+//   our class and interface with COM.
+// Arguments:
+// - pfnHandoff - Function to callback when a handoff is received
+// Return Value:
+// - S_OK, E_NOT_VALID_STATE (start called when already started) or relevant COM registration error.
+HRESULT CTerminalHandoff::s_StartListening(NewHandoff pfnHandoff) noexcept
+try
+{
+    RETURN_HR_IF(E_NOT_VALID_STATE, _pfnHandoff != nullptr);
+
+    auto classFactory = Make<SimpleClassFactory<CTerminalHandoff>>();
+
+    RETURN_IF_NULL_ALLOC(classFactory);
+
+    ComPtr<IUnknown> unk;
+    RETURN_IF_FAILED(classFactory.As(&unk));
+
+    RETURN_IF_FAILED(CoRegisterClassObject(__uuidof(CTerminalHandoff), unk.Get(), CLSCTX_LOCAL_SERVER, REGCLS_MULTIPLEUSE, &g_cTerminalHandoffRegistration));
+
+    _pfnHandoff = pfnHandoff;
+
+    return S_OK;
+}
+CATCH_RETURN()
+
+// Routine Description:
+// - Stops listening for TerminalHandoff requests by revoking the registration
+//   our class and interface with COM
+// Arguments:
+// - <none>
+// Return Value:
+// - S_OK, E_NOT_VALID_STATE (stop called when not started), or relevant COM class revoke error
+HRESULT CTerminalHandoff::s_StopListening() noexcept
+{
+    RETURN_HR_IF(E_NOT_VALID_STATE, _pfnHandoff == nullptr);

```

Comment:
Same here. There's no `RETURN_HR_IF_NULL`?

Reply:
This one actually exists. Changed.